misc/python/materialize/data_ingest/data_type.py

```diff
@@ -0,0 +1,187 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import random
+import string
+from enum import Enum
+from typing import Any
+
+
+class RecordSize(Enum):
+    TINY = 1
+    SMALL = 2
+    MEDIUM = 3
+    LARGE = 4
+
+
+class Backend(Enum):
+    AVRO = 1
+    POSTGRES = 2
+
+
+class DataType:
+    """As supported by Avro: https://avro.apache.org/docs/1.11.1/specification/_print/"""
+
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        raise NotImplementedError
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        raise NotImplementedError
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        raise NotImplementedError
+
+
+class NullType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        return None
+
+
+class BooleanType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        return random.choice((True, False))
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        return "boolean"
+
+
+class IntType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.randint(-127, 128)
+        elif record_size == RecordSize.SMALL:
+            return random.randint(-32768, 32767)
+        elif record_size == RecordSize.MEDIUM:
+            return random.randint(-2147483648, 2147483647)
+        elif record_size == RecordSize.LARGE:
+            return random.randint(-9223372036854775808, 9223372036854775807)
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
```

Comment:
I don't understand the purpose of this method. Does it always just return the input? 🤔

Reply:
A comment would be helpful. 😀

Reply:
Right, done. 

> Generate a value that corresponds to `num`, so that it will always be the same value for the same input `num`, but fits into the type. This doesn't make sense for a type like boolean.